Optimize SpatialBN on GPU

xiaomengy opened the pull request:
Summary: Optimize SpatialBN on GPU

Differential Revision: D13747581

BIT-silence commented:
Benchmark

SpatialBNGradient for N = 128, C = 256, H = 128, W = 128,
NCHW order: 140.17ms -> 49.4ms (CuDNN 55.6ms)
NHWC order: 624.14ms -> 128.14ms.

houseroad commented:
build still failed

xiaomengy committed:
Optimize SpatialBN on GPU (#16202)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16202

Optimize SpatialBN on GPU

Differential Revision: D13747581

fbshipit-source-id: 16dc2dd813915266e4cb662a1dc46635177e1b6d

houseroad reviewed (approved):
Looks good. I would recommend to add a brief description of the optimization in the PR summary as record.
